packages/autocomplete-plugin-algolia-insights/src/createViewedEvents.ts

```diff
@@ -0,0 +1,31 @@
+import { AlgoliaInsightsHit, ViewedObjectIDsParams } from './types';
+
+type CreateViewedEventsParams = {
+  items: AlgoliaInsightsHit[];
+};
+
+export function createViewedEvents({
+  items,
+}: CreateViewedEventsParams): Array<Omit<ViewedObjectIDsParams, 'eventName'>> {
+  const objectIDsByIndexName = items.reduce<Record<string, string[]>>(
+    (acc, current) => {
+      return {
+        ...acc,
```

Comment:
no need to spread acc in every iteration, you could set the data per index using .push and that should have a lower memory usage, since no many intermediary arrays & objects are created

Reply:
just suggesting since this in a slightly hotter path

Reply:
Adressed in 586c8c10a9de7da9614f7243c9ca874d26f354ab.